packages/arb-bridge-peripherals/contracts/tokenbridge/arbitrum/BuddyDeployer.sol

```diff
@@ -49,7 +52,19 @@ contract L2Deployer {
         }
 
         // L1 callback to buddy
-        bytes memory calldataForL1 = abi.encodeWithSelector(BuddyContract.finalizeBuddyDeploy.selector, !deployFail);
+        if(deployFail) {
+            bytes memory calldataForL1 = abi.encodeWithSelector(BuddyContract.finalizeBuddyDeploy.selector, false);
```

Comment:
Is there a reason to not have finalizeBuddyDeploy always called from the `L2Deployer
`?